protocol/auditlog.go

```diff
@@ -100,6 +116,7 @@ func (l ConiksAuditLog) Insert(addr string, signKey sign.PublicKey,
 	// create the new directory history
 	h = newDirectoryHistory(addr, signKey, snaps[0])
```

Comment:
Please see my comment: https://github.com/coniks-sys/coniks-go/pull/170#discussion_r132661384. I think we still need to verify `snaps[0]` when creating a new directory history. We could include this verification in `newDirectoryHistory`. What do you think?